quality_of_life/domain/quality_of_life_ranking.py

```diff
@@ -3,10 +3,17 @@
 from quality_of_life.repo.repository import read_dataset
 
 
-def get_city_ranking(limit: int) -> List[Dict]:
+def get_ranking_by(get_by, limit: int) -> List[Dict]:
     qol = read_dataset("quality_of_life_extended.csv")
-    ranking = [
-        {"city": element[0], "ranking": element[1]}
-        for element in list(zip(qol["UA_Name"], qol["mean"]))
-    ]
-    return sorted(ranking, key=lambda x: x["ranking"], reverse=True)[:limit]
+    ranking = {}
+    for element in list(zip(qol[get_by], qol["mean"])):
+        if element[0] not in ranking:
+            ranking[element[0]] = element[1]
+        else:
+            ranking[element[0]] += element[1]
```

Comment:
You could simplify your code with default dict:
https://www.geeksforgeeks.org/defaultdict-in-python/